bench/wtperf/config.c

```diff
@@ -652,6 +678,72 @@ config_sanity(CONFIG *cfg)
 	return (0);
 }
 
+void
+config_queue_dedupe(CONFIG *cfg)
+{
+	CONFIG_QUEUE_ENTRY *conf_line, *test_line, *tmp;
+	char *string_key;
+
+	conf_line = TAILQ_FIRST(&cfg->config_head);
+	while (conf_line != NULL) {
+		string_key = strchr(conf_line->string, '=');
+		tmp = test_line = TAILQ_NEXT(conf_line, c);
+		while (test_line != NULL) {
+			if (strncmp(conf_line->string, test_line->string,
+			    string_key - conf_line->string + 1) == 0) {
+				TAILQ_REMOVE(&cfg->config_head, conf_line, c);
+				free(conf_line->string);
+				free(conf_line);
+				break;
+			}
+			test_line = TAILQ_NEXT(test_line, c);
+		}
+		conf_line = tmp;
+	}
+}
+
+/*
+ * config_to_file --
+ *	Write the final config used in this execution to a file.
+ */
+void
+config_to_file(CONFIG *cfg)
+{
+	CONFIG_QUEUE_ENTRY *config_line;
+	FILE *fp;
+	size_t req_len;
+	char *path;
+
+	fp = NULL;
+
+	/* Backup the config */
+	req_len = strlen(cfg->home) + 100;
+	if ((path = calloc(req_len, 1)) == NULL) {
+		(void)enomem(cfg);
+		goto err;
+	}
+
+	snprintf(path, req_len + 14, "%s/CONFIG.wtperf", cfg->home);
+	if ((fp = fopen(path, "w")) == NULL) {
+		lprintf(cfg, errno, 0, "%s", path);
+		goto err;
+	}
+
+	/* Print the config dump */
+	fprintf(fp, "# Warning. This config can include defaults.\n");
+	fprintf(fp, "# This make cause differences in behaviour.\n");
```

Comment:
How about:
# Warning: This config includes unwritten, implicit configuration defaults.
# Changes to those values may cause differences in behavior.